### PR TITLE
Fix "access" fixtures and tests

### PR DESCRIPTION
**Pull request by guillep2k**
While working on #9787 I've noticed that some records in `access.yml` are off, because they do not match the values given by the rest of the fixture database.

Since `access` is only a derivative table, I've created a tool to re-create the corresponding fixture (`access.yml`) based on the content of the other tables. This revealed some inconsistences in the tests that I had to fix as well.

To re-create `access.yml` with the current algorithm from `access.go`, we can now use:

```
cd models
go test -tags "access_fixtures sqlite sqlite_unlock_notify" -run TestBuildAccessFixturesYaml
```

#### Summary of the modifications

- Added write collaborations for users 15 and 18 on repositories 21 and 22, since `access.yml` had those permissions but they were not backed by the rest of the database. Otherwise these permissions dissapeared and `TestSearchRepository()` failed at several points.
- Updated `TestUser_GetAccessibleRepositories()` to include repositories that user 2 had access to as well and should have been part of the count:
    - `repo3` (id:3): user is collaborator
    - `repo5` (id:5): user belongs to team id:1, with access to the repository
    - `repo21` (id:32): user belongs to team id:1, with access to the repository
- `access.yml` now contains records that are directly derived from the other `yml` files.
- `repo_unit.yml` include units for repo id:5, for which there were none.
- Fix `TestAPIOrgRepos` as now repo id:5 is visible to user id:2 as it should be.

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

File: models/fixture_access_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		fmt.Fprintf(w, "-\n")
+		fmt.Fprintf(w, "  id: %d\n", i+1)
+		fmt.Fprintf(w, "  user_id: %d\n", a.UserID)
+		fmt.Fprintf(w, "  repo_id: %d\n", a.RepoID)
+		fmt.Fprintf(w, "  mode: %d\n", a.Mode)
+		fmt.Fprintf(w, "\n")
+	}
+
+	w.Flush()
+}
```
Comment (lunny) [inline models/fixture_access_test.go:52]: f.Close()

Comment (guillep2k) [reply models/fixture_access_test.go:52]: @lunny Done.

**Comment by guillep2k** [comment]
Added:
- `repo_unit.yml` include units for repo id:5, for which there were none.
- Fix `TestAPIOrgRepos` as now repo id:5 is visible to user id:2 as it should be.

**Review by lunny — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Review by sapk — COMMENTED** [review]
It feel strange to have a test that generate file. It may be clearer to adding it to the standard tests suite and only check that the access table is valid ? Making the CI fail if it need to be updated ? And moving the generation in contrib/ maybe ?

**Comment by guillep2k** [comment]
> 
> 
> It feel strange to have a test that generate file. It may be clearer to adding it to the standard tests suite and only check that the access table is valid ? Making the CI fail if it need to be updated ? And moving the generation in contrib/ maybe ?

Yeah, I feel like that too. I'll work something out.

**Pull request merged**